paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java

```diff
@@ -77,6 +80,40 @@ public void createTag(Snapshot snapshot, String tagName) {
         }
     }
 
+    public void deleteTag(
+            String tagName, TagDeletion tagDeletion, SnapshotManager snapshotManager) {
+        checkArgument(!StringUtils.isBlank(tagName), "Tag name '%s' is blank.", tagName);
+        checkArgument(tagExists(tagName), "Tag '%s' doesn't exist.", tagName);
+
+        Snapshot taggedSnapshot = taggedSnapshot(tagName);
+        List<Snapshot> taggedSnapshots = taggedSnapshots();
```

Comment:
move `taggedSnapshots` down after `if (snapshotManager.snapshotExists)`.

Reply:
Cannot do it because delete the tag file will make the taggedSnapshots return wrong result.

Reply:
You can optimize it too:
```
if (snapshotManager.snapshotExists(taggedSnapshot.id())) {
    fileIO.deleteQuietly(tagPath(tagName)); 
    return;
} else {
    // document why this should be first.
    List<Snapshot> taggedSnapshots = taggedSnapshots();
    fileIO.deleteQuietly(tagPath(tagName)); 
}
```